### PR TITLE
BUG: Moment map retains WCS on write

### PR DESCRIPTION
**Pull request by pllim**
### Description

This pull request is to hook the Moment Map logic into our existing hack for everything else in Cubeviz.

### Change log entry

- [x] Is a change log needed? If yes, is it added to `CHANGES.rst`? If you want to avoid merge conflicts,
  list the proposed change log here for review and add to `CHANGES.rst` before merge. If no, maintainer
  should add a `no-changelog-entry-needed` label.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [x] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [x] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Did the CI pass? If not, are the failures related?
- [x] Is a milestone set? Set this to bugfix milestone if this is a bug fix and needs to be released ASAP; otherwise, set this to the next major release milestone.
- [ ] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)? [🐱](https://jira.stsci.edu/browse/JDAT-3766)

### CODE REVIEW

File: jdaviz/configs/cubeviz/plugins/moment_maps/moment_maps.py [hunk 1]
```diff
@@ -91,7 +91,11 @@ def calculate_moment(self, add_data=True):
             Whether to add the resulting data object to the app according to ``add_results``.
         """
         # Retrieve the data cube and slice out desired region, if specified
-        cube = self.dataset.get_object(cls=Spectrum1D, statistic=None)
+        if "_orig_spec" in self.dataset.selected_obj.meta:
+            cube = self.dataset.selected_obj.meta["_orig_spec"]
+        else:
+            cube = self.dataset.selected_obj
```
Comment (pllim) [inline jdaviz/configs/cubeviz/plugins/moment_maps/moment_maps.py:97]: This grabs the original data from collection and bypasses going through glue-astronomy translator. Should work unless there is something I overlooked? Maybe try it with a few different cubes and see.

File: CHANGES.rst [hunk 2]
```diff
@@ -147,6 +147,8 @@ API Changes
 Cubeviz
 ^^^^^^^
 
+- Fixes moment map losing WCS when being written out to FITS file. [#2431]
```
Comment (pllim) [inline CHANGES.rst:150]: Ooops I should probably move the change log entry.

**Comment by pllim** [comment]
Rebased to fix change log and pick up CI fixes from `main`. Original patch remains unchanged. FYI.

**Comment by pllim** [comment]
The test data Patrick gave me is about 50 MB. Maybe I can add a remote test...

**Review by kecnry — COMMENTED** [review]
Code changes make sense to me!  Agreed - either a remote test or with fake data would be great to get this new line covered and to catch any change to that internal metadata entry, etc.

File: jdaviz/configs/cubeviz/plugins/moment_maps/tests/test_moment_maps.py [hunk 3]
```diff
@@ -135,3 +140,17 @@ def test_write_momentmap(cubeviz_helper, spectrum1d_cube, tmp_path):
...
+    uri = "mast:jwst/product/jw02732-o003_t002_nirspec_prism-clear_s3d.fits"
+    download_path = str(tmp_path / Path(uri).name)
+    Observations.download_file(uri, local_path=download_path)
+
+    with warnings.catch_warnings():
+        warnings.simplefilter("ignore")
+        cubeviz_helper.load_data(download_path)
+    plugin = cubeviz_helper.plugins['Moment Maps']
+    plugin.calculate_moment()
+    assert isinstance(plugin._obj.moment.wcs, WCS)
```
Comment (pllim) [inline jdaviz/configs/cubeviz/plugins/moment_maps/tests/test_moment_maps.py:156]: I confirmed locally that this new test fails without the patch like this:
```
E       AssertionError: assert False
E        +  where False = isinstance(None, WCS)
```

Feel free to confirm it for yourself by running this test in a Jdaviz version without this patch.

Comment (kecnry) [reply jdaviz/configs/cubeviz/plugins/moment_maps/tests/test_moment_maps.py:156]: is it worth checking to make sure the WCS is actually the original?  Or is there no concern that that could ever not the be case and checking it exists is sufficient?

**Review by kecnry — APPROVED** [review]

**Comment by pllim** [comment]
> is it worth checking to make sure the WCS is actually the original?

It cannot be the original, strictly speaking. It would be the spatial component of the cube WCS. I guess does not hurt to add a check for the spatial part.

File: jdaviz/configs/cubeviz/plugins/moment_maps/tests/test_moment_maps.py [hunk 4]
```diff
@@ -135,3 +140,24 @@ def test_write_momentmap(cubeviz_helper, spectrum1d_cube, tmp_path):
...
+
+    # Because cube axes order is re-arranged by specutils on load, this gets confusing.
+    # There is a swapaxes within Moment Map WCS calculation.
+    sky_moment = plugin._obj.moment.wcs.pixel_to_world(50, 30)
+    sky_cube = cubeviz_helper.app.data_collection[0].meta["_orig_spec"].wcs.celestial.pixel_to_world(30, 50)  # noqa: E501
```
Comment (pllim) [inline jdaviz/configs/cubeviz/plugins/moment_maps/tests/test_moment_maps.py:161]: Note that Moment Map WCS takes `(x, y)` order but I have to swap that around to get the same sky coordinate from cube. Maybe @rosteen should double check if this is expected behavior given how specutils swap things and then we have to swap it back here and there.

File: jdaviz/configs/cubeviz/plugins/moment_maps/tests/test_moment_maps.py (hunk 3 above)
Comment (pllim) [reply jdaviz/configs/cubeviz/plugins/moment_maps/tests/test_moment_maps.py:156]: More added to test, see below.

File: jdaviz/configs/cubeviz/plugins/moment_maps/tests/test_moment_maps.py (hunk 4 above)
Comment (pllim) [reply jdaviz/configs/cubeviz/plugins/moment_maps/tests/test_moment_maps.py:161]: Given this line on L66 above:

```python
assert result.shape == (4, 2)  # Cube shape is (2, 2, 4)
```

Moment map dimension is `(ny, nx)` but looks like cube is `(nz, nx, ny)`. So maybe this makes sense? 🤪

**Comment by pllim** [comment]
Since there are 2 approvals and Patrick is confident the WCS is correct, I am merging. Thanks!

**Pull request merged**